pedersen/src/product_protocol.rs

```diff
@@ -0,0 +1,557 @@
+        inter: ProductProofIntermediate<P>,
+    ) -> ProductProofIntermediateTranscript<P> {
+        ProductProofIntermediateTranscript { 
+            alpha: inter.alpha,
+            beta: inter.beta,
+            delta: inter.delta,
+        }
+    }
+
+    /// make_transcript. This function simply adds `cx`, `cy`, `cxy` and `alpha`
+    /// `beta`, `delta` to the `transcript` object.
+    /// # Arguments
+    /// * `transcript` - the transcript which is modified.
+    /// * `cx` - the commitment to X that is being added to the transcript.
+    /// * `cy` - the commitment to Y that is being added to the transcript.
+    /// * `cxy` - the commitment to XY that is being added to the transcript.
+    /// * `alpha` - the alpha value that is being added to the transcript.
+    /// * `beta` - the beta value that is being added to the transcript.
+    /// * `delta` - the delta value that is being added to the transcript.
+    pub fn make_transcript(
+        transcript: &mut Transcript,
+        cx: &sw::Affine<P>,
+        cy: &sw::Affine<P>,
+        cxy: &sw::Affine<P>,
+        alpha: &sw::Affine<P>,
+        beta: &sw::Affine<P>,
+        delta: &sw::Affine<P>,
+    ) {
+        // This function just builds the transcript out of the various input 
+        // values. N.B Because of how we define the serialisation API to handle 
+        // different numbers, we use a temporary buffer here.
+        transcript.domain_sep();
+        let mut compressed_bytes = Vec::new();
+        cx.serialize_compressed(&mut compressed_bytes).unwrap();
+        transcript.append_point(b"CX", &compressed_bytes[..]);
+
+        cy.serialize_compressed(&mut compressed_bytes).unwrap();
+        transcript.append_point(b"CY", &compressed_bytes[..]);
+
+        cxy.serialize_compressed(&mut compressed_bytes).unwrap();
+        transcript.append_point(b"CXY", &compressed_bytes[..]);
+
+        alpha.serialize_compressed(&mut compressed_bytes).unwrap();
+        transcript.append_point(b"alpha", &compressed_bytes[..]);
+
+        beta.serialize_compressed(&mut compressed_bytes).unwrap();
+        transcript.append_point(b"beta", &compressed_bytes[..]);
+
+        delta.serialize_compressed(&mut compressed_bytes).unwrap();
+        transcript.append_point(b"delta", &compressed_bytes[..]);
+    }
+
+    /// create. This function returns a new product proof for `x`, `y` against 
+    /// `cx`, `cy` and `cxy`.
+    /// # Arguments
+    /// * `transcript` - the transcript object that is modified.
+    /// * `rng` - the RNG that is used to produce the random values. Must be 
+    /// cryptographically secure.
+    /// * `x` - the value that is used to show an opening of `cx`.
+    /// * `y` - the value that is used to show an opening of `cy`.
+    /// * `cx` - the commitment to x that is opened.
+    /// * `cy` - the commitment to y that is opened.
+    /// * `cxy` - the commitment to xy that is opened.
+    pub fn create<T: RngCore + CryptoRng>(
```

Comment:
`new` might be more idiomatic rust for functions returning `Self` but I don't feel strongly about it.